raspberry-pi/4/leds.nix

```
@@ -0,0 +1,116 @@
+{ config, lib, ... }:
+
+let
+  cfg = config.hardware.raspberry-pi."4".leds;
+in
+{
+  options.hardware = {
+    raspberry-pi."4".leds = {
+      disable-eth = lib.mkEnableOption ''
+        disable ethernet LEDs.
+      '';
+      disable-act = lib.mkEnableOption ''
+        disable activity LED.
+      '';
+      disable-pwr = lib.mkEnableOption ''
+        disable power LED.
```

mkEnableOption will add `Whether to enable X` as a description:

You could potentially just invert it by defining:

```
  mkDisableOption = name: mkOption {
    default = false;
    example = true;
    description = "Whether to disable ${name}.";
    type = lib.types.bool;
  };
```

```suggestion
      disable-eth = mkDisableOption ''ethernet LEDs.'';
      disable-act = mkDisableOption ''activity LED.'';
      disable-pwr = mkDisableOption ''power LED.'';
```

---

should we use `eth.disable` instead of `disable-eth`? what do you think? Just got the idea. Feels more like nix to me.

---

Applied your suggestions.

please resolve this thread, if you agree.